server_admin/topics/clients/oidc/con-secret-rotation.adoc

```diff
@@ -0,0 +1,28 @@
+[id="con-secret-rotation_{context}"]
+[[_secret_rotation]]
+= Client Secret Rotation
+[role="_abstract"]
+For client with <<_client-credentials, Confidential>> <<_access-type, Access Type>> {project_name} supports the functionality of rotating client secrets through <<_client_policies, Client Policies>>.
+
+The client secrets rotation policy provides greater security in order to alleviate problems such as secret leakage. Once enabled, {project_name} supports up to two concurrently active secrets for each client. The policy manages rotations according to the following settings:
+
+* *Secret expiration:* [seconds] - When the secret is rotated, this is the expiration of time of the new secret. The amount, _in seconds_, added to the secret creation date. Calculated at policy execution time.
+* *Rotated secret expiration:* [seconds] - When secret is rotated, this is the remaining expiration time for the old secret. This value should be always smaller than Secret expiration. When this is set to 0, the old secret will be immediately removed during client rotation. The amount, _in seconds_, added to the secret rotation date. Calculated at policy execution time.
+* *Remaining expiration time for rotation during update:* [seconds] - Time period when an update to a dynamic client should perform client secret rotation. Calculated at policy execution time.
+
+Whenever there is a client secret rotation a new main secret is generated and the old client main secret becomes the secondary secret with a new expiration date.
+
+== Rules for client secret rotation
+Rotations do not occur automatically or through a background process. In order to perform the rotation, an update action is required on the client, either through the {project_name} Admin Console UI or Admin REST API. When invoking a client update action, secret rotation occurs according to the rules:
+
+* When the value of *Secret expiration* is less than the current date.
+* During dynamic client registration client-update request, the client secret will be automatically rotated if the value of  *Remaining expiration time for rotation during update* match the period between the current date and the *Secret expiration*.
+
+Additionally it is possible through Admin REST API to force a client secret rotation at any time.
+
+[NOTE]
+====
+During the creation of new clients, if the client secret rotation policy is active, the behavior will be applied automatically.
+====
+
+WARNING: To apply the secret rotation behavior in pre-existing clients, before creating the policy, it is necessary that after defining the policy, an update must be performed on the client so that the behavior is applied.
```

Comment:
Similar to the comment above. I believe it can be "Re-generate secret" in case of admin console / admin REST API?

Reply:
You say, simplify the explanation and use the action of "Re-generate secret" by REST or Admin UI?

Reply:
@marcelomrwin Yes, exactly. That is what I meant.

Please correct me if I am wrong here: I assume example scenario when administrator did setup of the client policy with the executor and he wants to apply it on the existing clients through admin REST API / admin console. Then the administrator needs to:
- Go to some client in the admin console
- Go to tab "Credentials"
- Click "Re-generate secret" .
Am I correct here in the description of this procedure?

If yes, I suggest to mention that "Re-generate secret" is needed instead of just "Updating client" .

Updating client implies to me that administrator needs to do some change in the initial screen of the client (Tab "Details" of the client in the admin console) and click "Save" . But AFAIK updating client this way in the admin console is not sufficient to re-generate and rotate secret. Or am I wrong here?

Reply:
I chose to insert this information in the server_admin/topics/clients/oidc/proc-secret-rotation.adoc file because how it reports a process or a flow makes more sense.